### ISSUE TITLE
Allow expressions as default values in recipes

### ISSUE THREAD

**Issue by casey** [casey/just#391]
The following fails, but should probably work:

```
x = '2'

foo bar=x:
	echo {{bar}}
```

**Comment by runeimp** [comment]
This would make me so, very, very happy. 😃

**Comment by casey** [comment]
@runeimp Working on it!

Concatenation conflicts with variadic parameters, since they both use a `+`. This isn't a problem in languages where arguments are separated by commas, but since they aren't in just, the grammar of expressions conflicts with the grammar of function arguments.

The solution that I'm leaning towards is to only accept concatenations in default parameter expressions if they're inside of parenthesis.

So this would be okay:

```
foo y=("a" + "b"):
  echo {{y}}
```

```
$ just foo
echo ab
ab
```

But this would not be okay:

```
foo y="a" + "b":
  echo {{y}}
```

```
$ just foo
error: Expected name, but found '+'
  |
9 | foo y="a" + "b":
  |           ^
```

**Comment by casey** [comment]
Fixed in #400, cutting a release now!
